RawCMS.Library/Lambdas/JsDispatcher.cs

```diff
@@ -47,4 +48,102 @@ public override void Execute(string collection, ref JObject item, ref Dictionary
             }
         }
     }
+
+    public class PreSaveEvent : JsDispatcher
+    {
+        public override PipelineStage Stage => PipelineStage.PreOperation;
+
+        public override DataOperation Operation => DataOperation.Write;
+
+        public override string Name => "PreSaveEvent";
+
+        public override string Description => "PreSaveEvent";
+
+        protected readonly static EntityService entityService;
+
+        public PreSaveEvent(): base(entityService)
+        {
+            
+        }
+
+
+        public override void ExecuteInternal(string collection, ref JObject item, ref Dictionary<string, object> dataContext)
+        {
```

Comment:
this event is Execute. It call the ExecuteInternal of JsDispatcher, passing also Operation =Write and Stage=PreOperation, so that internal method cloud manage save.